src/test/groovy/life.qbic.samplestatus.reporter/ResultSpec.groovy

```diff
@@ -0,0 +1,120 @@
+package life.qbic.samplestatus.reporter
+
+import spock.lang.Specification
+
+import java.util.function.Function
+
+/**
+ * <class short description - 1 Line!>
+ *
+ * <More detailed description - When to use, what it solves, etc.>
+ *
+ * @since <version tag>
+ */
+class ResultSpec extends Specification {
```

Comment:
Very nice that you added those tests!
One behaviour is not covered. You state in the JavaDoc that whenever there is a value, no error must be present and vice versa. It would be very good to add a test for this as this is expected behaviour.

Reply:
Thanks @KochTobi , also tests for the map function are missing. I am already on it.